lua/magenta/actions.lua

```diff
@@ -47,14 +47,45 @@ local telescope_files = function()
   })
 end
 
+local snacks_files = function()
+  local snacks = require("snacks")
+
+  local completed = false
+  snacks.picker.pick({
+    source = "files",
+    title = "Select context files for Magenta",
+    confirm = function(picker)
+      if completed then return end
+      completed = true
+      local items = picker:selected({ fallback = true })
+
+      if #items > 0 then
+        local escaped_files = {}
+        for _, item in ipairs(items) do
+          table.insert(escaped_files, vim.fn.shellescape(item.file))
+        end
+        vim.cmd("Magenta context-files " .. table.concat(escaped_files, " "))
+      end
+
+      picker:close()
+    end,
+    on_close = function()
+      if completed then return end
+      completed = true
+    end,
+  })
+end
```

Comment:
I referenced this implementation: https://github.com/yetone/avante.nvim/blob/eb1cd44731783024621beafe4e46204cbc9a4320/lua/avante/ui/selector/providers/snacks.lua